webClient/src/app/editor/code-editor/monaco-settings/monaco-settings.component.ts

```diff
@@ -0,0 +1,252 @@
+    this.http.get<any>(ZoweZLUX.uriBroker.pluginConfigForScopeUri(this.pluginDefinition.getBasePlugin(),'user','monaco','editorconfig.json')).subscribe((response: any) => {
+      if (response.contents) {
+        this.config = response.contents.config;
+        this.jsonText = this.configToText();
+        this.setConfigFromJson();
+      } else {
+        this.initConfig();
+        this.jsonText = this.configToText();
+      }
+    },
+    //in case of error, just default                                                                                                                                            
+    (e:any)=> {
+      this.initConfig();
+      this.jsonText = this.configToText();
+    });
+  }
+
+  
+  
+  setConfig(attribute: string, value?: any, defaultValue?: any) {
+    let val = value !== undefined ? value : defaultValue;
+    let parts = attribute.split('.');
+    let parentObj = {};
+    let mirrored = true;
+    let configMirrorObj = this.config;
+    let lastObj = parentObj;
+    let currentObj = parentObj;
+    let pos = 0;
+    while (pos < parts.length) {
+      let newObj = {};
+      currentObj[parts[pos]] = newObj;
+      lastObj = currentObj;
+      currentObj = newObj;
+      if (mirrored) {
+        try {
+          configMirrorObj = configMirrorObj[parts[pos]];
+        } catch (e) {
+          mirrored = false;
+        }
+      }
+      pos++;
+    }
+    if (configMirrorObj && val === undefined) {
+      delete configMirrorObj[parts[parts.length-1]];
+    } else {
+      lastObj[parts[parts.length-1]] = val;
+      this.config = Object.assign(this.config, parentObj);
+    }
+  }
+
+  resetEditor() {
+    this.setConfigFromConfigService();
+    this.updateEditor();
+  }
+  
+  update(item: MonacoConfigItem) {
+    this.log.info('monaco config update item=%s, value=%s',item.attribute, item.value);
+    this.setConfig(item.attribute, item.value, item.default);
+    this.jsonText = this.configToText();
+    this.updateEditor();
+  }
+
+  private updateEditor() {
+    this.editor._themeService.setTheme(this.config.theme);
+    this.editor.updateOptions(this.config);
+    this.editor.setValue(this.jsonText);
+  }
+
+  updateFromPreview() {
+    let config = this.config;
+    try {
+      this.config = JSON.parse(this.editor.getValue());
+      this.jsonText = this.configToText();
+      this.updateEditor();
+      this.setConfigFromConfigService();
+    } catch (e) {
+      this.log.warn('Could not use preview JSON for config; Falling back to menu config');
+      this.config = config;
+      //ignore
+    }
+  }
+  
+  ngOnInit() {
+    this.editor = monaco.editor.create(this.monacoPreviewRef.nativeElement, this.config);
+    this.viewportEvents.resized.subscribe(()=> {
+      this.editor.layout()
+    });
+    monaco.editor.remeasureFonts();
+    this.setConfigFromConfigService();
+    setTimeout(()=> {
+      this.updateEditor();
+      this.editor.setModelLanguage('json');
```

Comment:
I get an error in the JS console: `setModelLanguage` is not a function. 
```typescript
this.editor.updateOptions({language: 'json'});
```
should work better.